docs/testplan/PFC_Congestion_Oversubscription_Test_Plan.md

```diff
@@ -0,0 +1,318 @@
+4.	Stop all traffic from the two IXIA ports 1 & 2
+5.	On the IXIA Rx port ensure that:
+a.	The received lossless traffic streams on IXIA port 3 show no missing and no out-of-order frames.
+b.	Lossy traffic streams rx rate should be equal to tx rate and did not drop any frames.
+6.	If possible, query the DUT drop counters to verify that no packets got discarded.
+7.	Repeat the test by adding a impairment (Network emulator) device between Tx port and DUT ingress port to simulate various cable lengths and validate that PFC headroom is sufficient for variable cable lengths.
+8.	Verify that the test results are same for varying cable lengths.
+
+
+### Test case # 2 – Many-to-One Congestion oversubscribe lossy traffic
+#### Test objective
+Test the response of lossy and lossless streams destined to a single egress port, with lossy traffic causing oversubscription of the egress port.
+PFC pause frames should automatically be generated by the device under test (DUT) to throttle back the lossless traffic sources causing the over-subscription, avoiding any lossless traffic drops.
+The two aspects that will be verified are whether:
+*	Packet drops really happen when lossy traffic suffers from congestion without triggering PFC
+*	WRR scheduler works as expected
+
+#### Background
+This test uses a similar diagram and description of the multi-ASIC router as the ‘Many-to-One Lossless Traffic Congestion’ test case.
+In this variant, the egress port packet forwarding engine, facing similar oversubscription demands as the previous test, will fairly grant resources to the VOQs in the packet forwarding engines.  The ingress port VOQs, with not enough allocation granted for the lossy traffic streams causing the congestion, will eventually run out of the memory buffer resources for the aggressor VOQs and will be forced drop some of their frames.
+
+#### Testbed Setup
+The test bed setup consists of three IXIA ports and a SONiC router as the device under test (DUT). All IXIA ports should have the same line speed.  The three DUT ports will be chosen at random across all the available ports in the router in order to eventually cover combinations of ports within the same line card and across line cards (or ports within the same ASIC vs. different ASICs on a single-card, multi-ASIC design).
+
+<p float="left">
+  <img src="Img/PFC_Testcase2.png" width="500"  hspace="250"/>
+</p>
+
+
+#### Test steps
+In this experiment we will create a total of four streams, with each of the two test ports sending two streams each. The aggregate bandwidth will be slightly over 100%:
+  *	The background traffic will include two lossless traffic streams, with the SONiC default lossless priorities of 3 and 4, and each having a 25% bandwidth.

```

Comment:
To avoid congestion on lossless priorities, can we reduce demand to 24% bandwidth?

Reply:
We tested with 25% bandwidth on lossless priorities and no congestion is observed.

Reply:
Done